pkg/hubble/exporter/dynamic_exporter.go

```diff
@@ -0,0 +1,161 @@
+	"github.com/cilium/cilium/pkg/hubble/exporter/exporteroption"
+
+	"github.com/cilium/cilium/api/v1/flow"
+	"github.com/cilium/cilium/pkg/lock"
+)
+
+const (
+	MAX_BACKUPS      = 1
+	MAX_FILE_SIZE_MB = 10
+)
+
+type dynamicExporter struct {
+	ctx              context.Context
+	logger           logrus.FieldLogger
+	watcher          *configWatcher
+	managedExporters map[string]*managedExporter
+	mutex            lock.RWMutex
+}
+
+func (d *dynamicExporter) OnDecodedEvent(ctx context.Context, event *v1.Event) (bool, error) {
+	d.mutex.RLock()
+	defer d.mutex.RUnlock()
+
+	for _, me := range d.managedExporters {
+		if me.config.End.After(time.Now()) {
+			me.exporter.OnDecodedEvent(ctx, event)
+		}
+	}
+	return false, nil
+}
+
+func NewDynamicExporter(configFilePath string) *dynamicExporter {
+	dynamicExporter := &dynamicExporter{
+		ctx:              context.TODO(),
+		logger:           logrus.New(),
+		managedExporters: make(map[string]*managedExporter),
+	}
+
+	watcher := NewConfigWatcher(configFilePath, dynamicExporter.onConfigReload)
+
+	dynamicExporter.watcher = watcher
+
+	return dynamicExporter
+}
+
+func (d *dynamicExporter) onConfigReload(config dynamicExportersConfig) {
+	d.mutex.Lock()
+	defer d.mutex.Unlock()
+
+	configuredFlowLogNames := make(map[string]bool)
+	for _, flowlog := range config.FlowLogs {
+		configuredFlowLogNames[flowlog.Name] = true
+		if _, ok := d.managedExporters[flowlog.Name]; ok {
+			d.applyUpdatedConfig(flowlog)
+		} else {
+			d.applyNewConfig(flowlog)
+		}
+	}
+
+	for flowLogName := range d.managedExporters {
+		if _, ok := configuredFlowLogNames[flowLogName]; !ok {
+			d.applyRemovedConfig(flowLogName)
+		}
+	}
+
+}
+
+func (d *dynamicExporter) applyNewConfig(flowlog flowLogConfig) {
+	exporterOpts := []exporteroption.Option{
+		exporteroption.WithPath(flowlog.FilePath),
+		exporteroption.WithMaxSizeMB(MAX_FILE_SIZE_MB),
+		exporteroption.WithMaxBackups(MAX_BACKUPS),
+		exporteroption.WithAllowList(flowlog.IncludeFilters),
+		exporteroption.WithDenyList(flowlog.ExcludeFilters),
+		exporteroption.WithFieldMask(flowlog.FieldMask),
+	}
+
+	exporter, err := NewExporter(d.ctx, d.logger.WithField("flowLogName", flowlog.Name), exporterOpts...)
+
+	if err != nil {
+		d.logger.Errorf("Failed applying flowlog for name: %s; %v", flowlog.Name, err)
+	}
+
+	d.managedExporters[flowlog.Name] = &managedExporter{
+		config:   flowlog,
+		exporter: exporter,
+	}
+
+}
+
+func (d *dynamicExporter) applyUpdatedConfig(flowlog flowLogConfig) {
+	m := d.managedExporters[flowlog.Name]
+	if !configChanged(m.config, flowlog) {
+		return
+	}
+	d.applyRemovedConfig(flowlog.Name)
+	d.applyNewConfig(flowlog)
+}
+
+func configChanged(configA flowLogConfig, configB flowLogConfig) bool {
+	if configA.FilePath != configB.FilePath {
+		return true
+	}
+
+	if !reflect.DeepEqual(configA.FieldMask, configB.FieldMask) {
+		return true
+	}
+
+	if !compareFilters(configA.IncludeFilters, configB.IncludeFilters) {
```

Comment:
Perhaps If we were to define a filter collection type, we could then hang `Equal` method on that type?

```go
type FlowFilters []*flowpb.FlowFilter 
```